internal/fabric/client/client_ccp.go

```diff
@@ -0,0 +1,244 @@
+	"github.com/hyperledger/fabric-sdk-go/pkg/client/channel"
+	"github.com/hyperledger/fabric-sdk-go/pkg/client/channel/invoke"
+	"github.com/hyperledger/fabric-sdk-go/pkg/common/errors/retry"
+	"github.com/hyperledger/fabric-sdk-go/pkg/common/providers/context"
+	"github.com/hyperledger/fabric-sdk-go/pkg/common/providers/core"
+	"github.com/hyperledger/fabric-sdk-go/pkg/common/providers/fab"
+	"github.com/hyperledger/fabric-sdk-go/pkg/common/providers/msp"
+	"github.com/hyperledger/fabric-sdk-go/pkg/core/cryptosuite"
+	"github.com/hyperledger/fabric-sdk-go/pkg/fabsdk"
+	mspImpl "github.com/hyperledger/fabric-sdk-go/pkg/msp"
+	log "github.com/sirupsen/logrus"
+)
+
+// the ccpClientWrapper implements the RPCClient interface using the fabric-sdk-go implementation
+// based on the static network description provided via the CCP yaml
+type ccpClientWrapper struct {
+	channelClient   *channel.Client
+	channelProvider context.ChannelProvider
+	signer          *msp.IdentityIdentifier
+}
+
+// defined to allow mocking in tests
+type channelCreator func(context.ChannelProvider) (*channel.Client, error)
+
+type ccpRPCWrapper struct {
+	txTimeout           int
+	sdk                 *fabsdk.FabricSDK
+	cryptoSuiteConfig   core.CryptoSuiteConfig
+	userStore           msp.UserStore
+	idClient            IdentityClient
+	ledgerClientWrapper *ledgerClientWrapper
+	eventClientWrapper  *eventClientWrapper
+	channelCreator      channelCreator
+	// one channel client per channel ID, per signer ID
+	channelClients map[string](map[string]*ccpClientWrapper)
+}
+
+func newRPCClientFromCCP(configProvider core.ConfigProvider, txTimeout int, userStore msp.UserStore, idClient IdentityClient, ledgerClientWrapper *ledgerClientWrapper, eventClientWrapper *eventClientWrapper) (RPCClient, error) {
+	configBackend, _ := configProvider()
+	cryptoConfig := cryptosuite.ConfigFromBackend(configBackend...)
+	identityConfig, err := mspImpl.ConfigFromBackend(configBackend...)
+	if err != nil {
+		return nil, errors.Errorf("Failed to load identity configurations: %s", err)
+	}
+	clientConfig := identityConfig.Client()
+	if clientConfig.CredentialStore.Path == "" {
+		return nil, errors.Errorf("User credentials store path is empty")
+	}
+
+	log.Infof("New gRPC connection established")
+	w := &ccpRPCWrapper{
+		sdk:                 ledgerClientWrapper.sdk,
+		cryptoSuiteConfig:   cryptoConfig,
+		userStore:           userStore,
+		idClient:            idClient,
+		ledgerClientWrapper: ledgerClientWrapper,
+		eventClientWrapper:  eventClientWrapper,
+		channelClients:      make(map[string]map[string]*ccpClientWrapper),
+		channelCreator:      createChannelClient,
+		txTimeout:           txTimeout,
+	}
+	return w, nil
+}
+
+func (w *ccpRPCWrapper) Init(channelId, signer, chaincodeName, method string, args []string) (*TxReceipt, error) {
+	log.Tracef("RPC [%s:%s:%s] --> %+v", channelId, chaincodeName, method, args)
```

Comment:
Should the trace message distinguish between `Init` and `Invoke`? Won't be obvious when looking at logs, I assume

Reply:
Based on discussion with @jimthematrix , he clarified that `Init()` is signified with a flag in the payload. We agreed that the internal interface could eliminate separate `Init()` & `Invoke()` and use only one.

Reply:
good suggestion. per our offline discussion I eliminated the `Init()` method from the `RPCClient` interface that's only artificially different than the `Invoke()` method, and added `isInit` to the logging